xtask/src/ci/target_matrix.rs

```diff
@@ -0,0 +1,152 @@
+use clap::builder::BoolishValueParser;
+use clap::Parser;
+use serde::Serialize;
+
+use crate::util::{get_matrix, gha_output, gha_print, CiTarget, ImageTarget};
+
+pub(crate) fn run(message: String, author: String) -> Result<(), color_eyre::Report> {
+    let mut matrix: Vec<CiTarget> = get_matrix().clone();
+    if author == "bors[bot]" && message.starts_with("Try #") {
+        if let Some((_, args)) = message.split_once(": ") {
+            let app = TargetMatrixArgs::parse_from(args.split(' '));
+            app.filter(&mut matrix);
+        }
+    } else {
+        gha_print("Running all targets.");
+    }
+
+    let matrix = matrix
+        .iter()
+        .map(|target| Matrix {
+            pretty: target.to_image_target().alt(),
+            platforms: target.platforms(),
+            target: &target.target,
+            sub: target.sub.as_deref(),
+            os: &target.os,
+            run: target.run.map(|b| b as u8),
+            deploy: target.deploy.map(|b| b as u8),
+            build_std: target.build_std.map(|b| b as u8),
+            cpp: target.cpp.map(|b| b as u8),
+            dylib: target.dylib.map(|b| b as u8),
+            runners: target.runners.as_deref(),
+            std: target.std.map(|b| b as u8),
+        })
+        .collect::<Vec<_>>();
+    gha_output("matrix", &serde_json::to_string(&matrix)?);
+    Ok(())
+}
+
+#[derive(Serialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+struct Matrix<'a> {
```

Comment:
Can we change the naming on this maybe? `Matrix` and `MATRIX` was somewhat confusing early, since the former was a specific element and the `MATRIX` was the `Vec<Matrix>`. Normally the matrix runs every combination, and we just create a single set of combinations (which absolutely works). `MatrixElement` could work, even though it's not ideal, but it's much clearer than `Matrix`. That way we don't use matrix for both a list of elements and the element itself.

I understand for serialization why this differs.